### ISSUE TITLE
Placeholder: Support Shanghai in Retesteth

### ISSUE THREAD

**Issue by siladu** [hyperledger/besu#4885]
Placeholder issue to save some info and track, maybe end up being more than one issue as I'm not sure of the scope of this yet.

https://github.com/ethereum/tests is being updated for Shanghai.

Dimitry (from ethereum/tests) asked about besu support: https://discord.com/channels/905194001349627914/1050442708587647006/1057617212925554779
> Besu client support test rpcs, however test_setChainParams does not trigger Merge rulset activation, how do I make it run in Merge mode?
it should parse the content of request the same way it is defined in chainparams besu config file when starting the client and defining fork blocks

Some work has been done by @shemnon on https://github.com/shemnon/besu/tree/t8n "to provide t8n access to the retesteth services"

but retesteth harness/setChainParams in besu is currently flakey and needs looking at.

---

Some notes from @garyschulte about running retesteth:

I don’t think retesteth has RPC specific instructions anymore, and their baseline howto is geth centric, but it covers the necessary steps to setup a test context:
https://ethereum-tests.readthedocs.io/en/latest/retesteth-tutorial.html#retesteth-in-a-docker-container

retesteth works with besu via rpc interface:
https://ethereum-tests.readthedocs.io/en/latest/rpc-ref.html?highlight=besu

in a nutshell, it is easiest to run besu outside of the retesteth docker conatiner, on a specific port, and point retesteth to it:
start besu (possibly from intellij),  e.g.:

`besu retesteth --rpc-http-port 47710 --rpc-http-host="192.168.0.108" --host-allowlist=all --data-path=/tmp/retesteth --logging=DEBUG`

and then run retesteth against it, e.g.:

`sh ./dretesteth.sh -t GeneralStateTests -- --all --lowcpu -j1 --nodes 192.168.0.108:47710 --testpath `pwd` --datadir tests/config3 --clients besu`

**Comment by non-fungible-nelson** [comment]
Closing for now @siladu
